streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamKStreamJoin.java

```diff
@@ -200,11 +217,11 @@ private void emitNonJoinedOuterRecords(final KeyValueStore<TimestampedKeyAndJoin
             // throttle the emit frequency to a (configurable) interval;
             // we use processing time to decouple from data properties,
             // as throttling is a non-functional performance optimization
-            if (context.currentSystemTimeMs() < sharedTimeTracker.nextTimeToEmit) {
+            if (internalProcessorContext.currentSystemTimeMs() < sharedTimeTracker.nextTimeToEmit) {
```

Comment:
@vvcephei, I haven't found a way to get the current system time. Is there a better alternative than this?